### DISCUSSION TITLE
root query context

### DISCUSSION THREAD

**Discussion by JonathanFraser** [mirumee/ariadne#842] [Q&A]
I'm implementing tracing on our graphql server but I'm getting a bit stuck. What I have currently is a custom extension to inject datadog spans on each resolver. This works nicely and we can see some good data. However, each of the resolvers are in parallel and more importantly independent. The resolvers aren't run in their parent's context. What I'd like to be able to do is hook into some sort of context which doesn't finish until all children are resolved, that way I can get a better feel for total root query execution time.

**Comment by rafalp** [comment]
There's no such context in query executor because as you've found out, it doesn't really execute queries recursively, doing it in layers instead. This behaviour is not really specific to Ariadne but is part of `graphql-core` library which is python port of graphql.js

Our OpenTracing extension relies on `info.path` thats passed to every resolver to know where in GraphQL query this resolver is and to name spans in a way that produces useable output in Jaeger. I have no experience with DataDog but maybe that will be also useful to you?
